internal/cluster/service.go

```diff
@@ -201,6 +201,9 @@ type Service interface {
 
 	// DeleteNodePool deletes a node pool from a cluster.
 	DeleteNodePool(ctx context.Context, clusterID uint, name string) (deleted bool, err error)
+
+	// List NodePools
+	ListNodePools(ctx context.Context, clusterID uint) ([]NodePool, error)
```

Comment:
Not sure if you can return this NodePool type here, I'm afraid we will have to use `map[string]interface{}`